valhalla/baldr/time_info.h

```diff
@@ -276,6 +281,19 @@ struct TimeInfo {
               << ", negative_seconds_from_now: " << ti.negative_seconds_from_now
               << ", tz_cache: " << ti.tz_cache << "}";
   }
+
+  // for unit tests
+  TimeInfo(uint64_t _valid = false,
+           uint64_t _timezone_index = 0,
+           uint64_t _local_time = 0,
+           uint64_t _second_of_week = 0,
+           uint64_t _seconds_from_now = 0,
+           uint64_t _negative_seconds_from_now = false,
+           baldr::DateTime::tz_sys_info_cache_t* _tz_cache = nullptr)
+      : valid(_valid), timezone_index(_timezone_index), local_time(_local_time),
+        second_of_week(_second_of_week), seconds_from_now(_seconds_from_now),
+        negative_seconds_from_now(_negative_seconds_from_now), tz_cache(_tz_cache) {
+  }
```

Comment:
This is actually a constructor for the unit tests for lines like A2 bellow (`baldr::TimeInfo{false}`) where not all the field's values are given, the compiler complains about missing the rest of the values.
On line A1 (`baldr::TimeInfo{false, 0, 0, 0, 0, 0, cache}`) it does not complain because everything is given.

Example
On [line 114 of test/gurka/test_time_tracking.cc](https://github.com/valhalla/valhalla/blob/master/test/gurka/test_time_tracking.cc#L114-L116)
```
    // invalid should stay that way                                                                                     
    auto ti = baldr::TimeInfo{false, 0, 0, 0, 0, 0, cache}.forward(10, 1);                     //A1                         
    ASSERT_EQ(ti, (baldr::TimeInfo{false}));                                                            //A2
```
There are zillions of lines like A2
From my point of view it is cleaner to have a constructor than doing zillions of changes like
```
ASSERT_EQ(ti, (baldr::TimeInfo{false,0, 0, 0, 0, 0, cache}));   
```
That is why I also added the comment of "for unit tests"

Reply:
yeah I thought so, I'm not arguing we shouldn't have that constructor. I was just suggesting to not put it here but make it a constructor of a class `TimeInfoTest` which we can put in the test cc file and subclass from `TimeInfo`. since we only need this constructor in the tests. but either way, it was just a suggestion, I'm not very partial to keeping it here, it's still useful even we don't use it in the library atm.

Reply:
actually I'm getting more partial to my suggestion:) can we do what I suggested?